### ISSUE TITLE
Newer version of Odoo

### ISSUE THREAD

**Issue by norlinhenrik** [cetmix/cetmix-tower#110]
@ivs-cetmix 

When would you like to upgrade `cetmix_tower_server`, and to which version (15/16/17/18)? I think it is good to set a date where the pull requests for 14.0 should be done so we can move on to a supported verison. Or do you think we will have to develop in parallel on more than one version?

**Comment by ivs-cetmix** [comment]
Hi @norlinhenrik, this was announced on our OCA Talks when we open sourced the project. You can check link to the talk and the PDF version of the presentation on the Cetmix Tower official page: https://cetmix.com/tower

To sum it up:

- Release for 14.0: November 2024
- Migration to 16.0: Planned for 2024
- Migration to 18.0: next year
- 15.0/17.0 and other odd versions are planned only if we will have time. We are planning to focus on even versions primarily. However we welcome migration PR's done by community members. As long as those PR's follow the OCA standards.

**Comment by badbole** [comment]
Hi.. i just tested quick port n 16.0 
you can check it here: 
https://github.com/badbole/cetmix-tower/tree/16.0-dev

basicly, all  works fine, someJS needs rewrite... but i can use it.. 
i added some minor thingies in dev branch.. but hey.. .it works fine and i'm starting to get familiar and use it on 16

**Comment by ivs-cetmix** [comment]
> Hi.. i just tested quick port n 16.0 you can check it here: https://github.com/badbole/cetmix-tower/tree/16.0-dev
> 
> basicly, all works fine, someJS needs rewrite... but i can use it.. i added some minor thingies in dev branch.. but hey.. .it works fine and i'm starting to get familiar and use it on 16

Hi @badbole thanks a lot for your contribution! 
Yup, there shouldn't be too much to do, however besides the JS we should update the views and also rewrite the tests to replace "setUp" with "setUpClass".
This week we are planning to merge some important changes for the SSH part, including this PR: https://github.com/cetmix/cetmix-tower/pull/206
After that we will open the final PR to merge 14.0-dev into 14.0 and next week (hope so 😄 ) we will start the migration process.

P.S. I would appreciate your review here too: https://github.com/cetmix/cetmix-tower/pull/206

**Comment by badbole** [comment]
@ivs-cetmix thany you and Cetmix for this nie solution!
I have checked and adopted views (leaving comments why in my code)
Also.. i have noticed a few possible improovment places, like:

In server template: 
Variables coud have sequence ( ordering posibility) as well as selection instead on boolen for required...
Also section/commnet part (simmilar to sale order line, invoice line) would be nice to have...
Curently we can require variable or leave it empty...

But it would be nice if we could conditionaly require variable... so:
required_selection could be: 
No- not required
Yes - always required
Conditionaly - required if the value of previous set variable matches condition...

Real life example:
In flight pplan to setup instance we could have 

variable: Postgres server , with options: Install new | Use existing...

folowing that... variables like: postgres_server, postgres_port, postgres_username, postgres_password (should be a secret)
would be required if we use existing server,
or left empty if se install new postgres server and set it up with Flight Plan...


I have some more ideas... but i belive i should open them as separate issues?

p.s. left some comments on #206 ... good work i was testing exactly that parts recenty

**Comment by ivs-cetmix** [comment]
> [@ivs-cetmix](https://github.com/ivs-cetmix) thany you and Cetmix for this nie solution! I have checked and adopted views (leaving comments why in my code) Also.. i have noticed a few possible improovment places, like:
> 
> In server template: Variables coud have sequence ( ordering posibility) as well as selection instead on boolen for required... Also section/commnet part (simmilar to sale order line, invoice line) would be nice to have... Curently we can require variable or leave it empty...
> 
> But it would be nice if we could conditionaly require variable... so: required_selection could be: No- not required Yes - always required Conditionaly - required if the value of previous set variable matches condition...
> 
> Real life example: In flight pplan to setup instance we could have
> 
> variable: Postgres server , with options: Install new | Use existing...
> 
> folowing that... variables like: postgres_server, postgres_port, postgres_username, postgres_password (should be a secret) would be required if we use existing server, or left empty if se install new postgres server and set it up with Flight Plan...
> 
> I have some more ideas... but i belive i should open them as separate issues?
> 
> p.s. left some comments on [#206](https://github.com/cetmix/cetmix-tower/pull/206) ... good work i was testing exactly that parts recenty

Regarding conditionally based required we also consider this feature. So this is something to be implemented. 
And yes, we really appreciate your feedback, so opening a feature request is something that would be nice too!
There are a lot of cool features in the backlog already, and we will start adding them as soon as the projects gets live officially.
